airflow/providers/google/common/deprecated.py

```diff
@@ -0,0 +1,176 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import inspect
+import re
+from datetime import date, datetime
+from typing import Any, Callable
+
+from deprecated import deprecated as standard_deprecated
+from deprecated.classic import ClassicAdapter
+
+
+class AirflowDeprecationAdapter(ClassicAdapter):
```

Comment:
Just to clarify...
I thought we are after trying to reduce the code in `providers.google.common` as it makes it harder to split the google provider.

Do you really want to add this complex interface which may force you to copy-paste it to all google "sub" providers?

Reply:
It can stay here for a while as an experimental feature because it is now needed not only in google.cloud. And during stage 2 (quotes below) we will move it to another location

> Stage 2. Learn from it and move it (possibly extending this to other cases like version mentioned by @dstandish ) -> either to common.compat or future airflow-sdk if it's ready and depended on by providers by that time.

Reply:
@eladkal , hi
Can we merge it please if you don't have other questions?